app/controllers/events_controller.rb

```diff
@@ -1,12 +1,13 @@
 class EventsController < ApplicationController
-  before_action :authenticate_user!, only: [:new, :create, :edit, :update, :destroy]
+  before_action :authenticate_user!, only: [:index, :new, :create, :edit, :update, :destroy]
```

Comment:
since you have not authenticate_user when action is show.
`before_action :set_current_event` can not get current_user